### PR TITLE
Use provider schema Terraform CLI

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

This PR adds make targets and makes other required changes to generate types using the provider schema from Terraform CLI instead of importing the source code of Terraform providers.

Also makes couple of improvements:

- Updates go version to 1.17
- Adds `build` directory as a submodule
- Bump Terraform CLI to latest (`1.1.4`)
- Use [null provider](https://registry.terraform.io/providers/hashicorp/null/3.1.0) instead of `hashicups` to have code compiled.
- Other small improvements...

As part of https://github.com/crossplane/terrajet/issues/175

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Follow the updated guide from https://github.com/crossplane/terrajet/pull/174 

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,22 +1,102 @@
 module github.com/crossplane-contrib/provider-jet-template
 
-go 1.16
+go 1.17
+
+replace github.com/crossplane/terrajet => github.com/turkenh/terrajet v0.0.0-20220124141610-617a1c30a0dc
```
Comment (turkenh) [inline go.mod:5]: To be removed when https://github.com/crossplane/terrajet/pull/174 merged

File: config/schema.json [hunk 2]
```diff
@@ -0,0 +1 @@
...n a future version.","description_kind":"markdown","deprecated":true,"computed":true},"inputs":{"type":["map","string"],"description":"A map of arbitrary strings that is copied into the `outputs` attribute, and accessible directly for interpolation.","description_kind":"markdown","optional":true},"outputs":{"type":["map","string"],"description":"After the data source is \"read\", a copy of the `inputs` map.","description_kind":"markdown","computed":true},"random":{"type":"string","description":"A random value. This is primarily for testing and has little practical use; prefer the [hashicorp/random provider](https://registry.terraform.io/providers/hashicorp/random) for more practical random number use-cases.","description_kind":"markdown","computed":true}},"description":"The `null_data_source` data source implements the standard data source lifecycle but does not\ninteract with any external APIs.\n\nHistorically, the `null_data_source` was typically used to construct intermediate values to re-use elsewhere in configuration. The\nsame can now be achieved using [locals](https://www.terraform.io/docs/language/values/locals.html).\n","description_kind":"markdown","deprecated":true}}}}}}
```
Comment (muvaf) [inline config/schema.json:1]: Is there an easy way to make this look formatted so that it's easier to debug?

File: build/CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -1,3 +0,0 @@
-## Community Code of Conduct
```
Comment (muvaf) [inline build/CODE_OF_CONDUCT.md:1]: Is this intentional?

**Review by muvaf — APPROVED** [review]
LGTM! Thanks @turkenh !

File: config/schema.json (hunk 2 above)
Comment (turkenh) [reply config/schema.json:1]: Yeah, this was also something I tried out but couldn't find an easy way to do so (without introducing a new dependency like `jq` / `python`). 

Personally, I format it using my IDE if/when I need to debug.

File: build/CODE_OF_CONDUCT.md (hunk 3 above)
Comment (turkenh) [reply build/CODE_OF_CONDUCT.md:1]: Yes, the build folder was previously not a `submodule`. This PR deletes old folder and adds it as a submodule (see the PR description)

**Pull request merged**